src/providers/wfs/qgswfssourceselect.cpp

```diff
@@ -333,6 +333,16 @@ void QgsWFSSourceSelect::oapifLandingPageReplyFinished()
 
   mAvailableCRS.clear();
   QString url( mOAPIFLandingPage->collectionsUrl() );
+
+  // Add back any extra query parameters, see issue GH #46535
+  const QgsWfsConnection connection( cmbConnections->currentText() );
+  const QUrl connectionUrl( connection.uri().param( QStringLiteral( "url" ) ) );
+  if ( ! connectionUrl.query().isEmpty() )
+  {
+    url.append( '?' );
```

Comment:
it might be prudent to test if url has already query parameters, and append '&' instead of '?' in that situation

Reply:
I don't think it's necessary, because the url coming from `mOAPIFLandingPage->collectionsUrl()` has already stripped the query string (which was the reason for the bug).

See: https://github.com/qgis/QGIS/pull/47145/files#diff-80852703dcc5dfa32d7ec9d31e4170a03146d6787bf637444e54bf9d18c61362R138

Reply:
hum, I'm not sure why we strip off query parameter from the URL. Maybe we should keep them ? There was perhaps a reason for doing that, but of course I've already forgotten...

Reply:
I think it's because the collections url is used a base for appending the individual collection urls, my first attempt was to comment out the query string stripping block but the tests were not passing due to the fact that the individual collections urls were appended after the query string, something like `collections?my_param=xxx/my_collection`. By not altering the url returned from `mOAPIFLandingPage->collectionsUrl()` and appending it here we avoid that problem.

Reply:
> I think it's because the collections url is used a base for appending the individual collection urls

ah right